var/spack/repos/builtin/packages/py-dace/package.py

```diff
@@ -0,0 +1,54 @@
+# Copyright 2013-2023 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.package import *
+
+
+class PyDace(PythonPackage):
+    """DaCe is a fast parallel programming framework that takes code in
+    Python/NumPy and other programming languages, and maps it to
+    high-performance CPU, GPU, and FPGA programs, which can be
+    optimized programmatically or interactively."""
+
+    homepage = "https://github.com/spcl/dace"
+    pypi = "dace/dace-0.15.1.tar.gz"
+    git = "https://github.com/spcl/dace.git"
+
+    maintainers("tbennun")
+
+    version("master", branch="master", submodules=True)
+    version("0.15.1", sha256="69bfdbbd5c7177f2926a874f5fa82fcdef61fc532c022b4bc12e1e9218724093")
+
+    variant(
+        "counters",
+        description="Optional requirements that enable performance counter collection.",
+        default=False,
+    )
+
+    depends_on("python@3.6:3.12", type=("build", "run"))
+    depends_on("py-setuptools", type="build")
+
+    # Dependencies from setup.py
+    depends_on("py-numpy@1.21:", type=("build", "run"))
```

Comment:
Personally I wouldn't trust requirements.txt for versions

Reply:
Fixed